cli/bpmetadata/tfconfig.go

```diff
@@ -0,0 +1,257 @@
+package bpmetadata
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+)
+
+const (
+	versionRegEx = "/v([0-9]+[.0-9]*)$"
+)
+
+type blueprintVersion struct {
+	moduleVersion   string
+	requiredVersion string
+}
+
+var rootSchema = &hcl.BodySchema{
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type:       "terraform",
+			LabelNames: nil,
+		},
+		{
+			Type:       "locals",
+			LabelNames: nil,
+		},
+		{
+			Type:       "resource",
+			LabelNames: []string{"type", "name"},
+		},
+	},
+}
+
+var metaSchema = &hcl.BodySchema{
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type:       "provider_meta",
+			LabelNames: []string{"name"},
+		},
+	},
+}
+
+var metaBlockSchema = &hcl.BodySchema{
+	Attributes: []hcl.AttributeSchema{
+		{
+			Name: "module_name",
+		},
+	},
+}
+
+// getBlueprintVersion gets both the required core version and the
+// version of the blueprint
+func getBlueprintVersion(configPath string) *blueprintVersion {
+	bytes, err := os.ReadFile(configPath)
+	if err != nil {
+		return nil
+	}
+
+	//create hcl file object from the provided tf config
+	fileName := filepath.Base(configPath)
+	var diags hcl.Diagnostics
+	p := hclparse.NewParser()
+	versionsFile, fileDiags := p.ParseHCL(bytes, fileName)
+	diags = append(diags, fileDiags...)
```

Comment:
I thought about that but in that case the execution won't go too far since the file wouldn't have any blocks to parse and the two fns below would just exit out quickly returning nils. Anyway I created a fn for checking errors, that way we can keep this check DRY